### PR TITLE
api: add namespace arg on getProcessList

### PR DESCRIPTION
**Pull request by jordipainan**
- The namespace of a processes contract can be passed
  as an argument to the getProcessList request.
  This allows to fetch only the processes that are registred
  on a specific namespace and not all the processes of the vochain
  for a given entity.
- The 0 value is valid and represents the namespace 0.
- Is possible to obtain all the processes of all namespaces
  just by not passing the namespace argument on the request at all.

### CODE REVIEW

File: vochain/scrutinizer/process.go [hunk 1]
```diff
@@ -37,8 +37,18 @@ func (s *Scrutinizer) ProcessList(entityID []byte, fromID []byte, max int64) ([]
 		if !fromLock {
 			keyCopy := make([]byte, len(process))
 			copy(keyCopy, process)
-			processListResult = append(processListResult, keyCopy)
-			max--
+			if p, err := s.VochainState.Process(keyCopy, true); err != nil {
+				log.Debugf("error fetching process %x: %s", p.ProcessId, err)
+				continue
+			} else if namespace == nil {
```
Comment (p4u) [inline vochain/scrutinizer/process.go:43]: There is no need to do an else here if there is a `continue` on the line above.

File: vochain/scrutinizer/process.go [hunk 2]
```diff
@@ -37,8 +37,18 @@ func (s *Scrutinizer) ProcessList(entityID []byte, fromID []byte, max int64) ([]
 		if !fromLock {
 			keyCopy := make([]byte, len(process))
 			copy(keyCopy, process)
-			processListResult = append(processListResult, keyCopy)
-			max--
+			if p, err := s.VochainState.Process(keyCopy, true); err != nil {
```
Comment (p4u) [inline vochain/scrutinizer/process.go:40]: I don't like this approach, is very expensive. For each ProcessList query we'll be fetching all processes from the state...
I'd rather try to save on the processList scrutinizer storage the namespace. So splinting by bytes (reserving N bytes for the namespace) would make the trick.

File: vochain/scrutinizer/process.go [hunk 3]
```diff
@@ -37,8 +37,18 @@ func (s *Scrutinizer) ProcessList(entityID []byte, fromID []byte, max int64) ([]
...
+				log.Debugf("error fetching process %x: %s", p.ProcessId, err)
+				continue
```
Comment (p4u) [inline vochain/scrutinizer/process.go:42]: If there is a continue here the else is not required....

**Review by p4u — CHANGES_REQUESTED** [review]
I can work on this if you want. Just let me know and I'll take the work from your current branch.

**Comment by jordipainan** [comment]
Yes ofc, feel free to continue the work from here.
Just to point it out, any solution will be a tmp workaround since the right solution is to not use the scrutinizer for getting the entities processes but the process tree (indexing each process by its entityid)

File: vochain/scrutinizer/process.go (hunk 1 above)
Comment (jordipainan) [reply vochain/scrutinizer/process.go:43]: the else is required for accessing p afterwards. If you do not want the else here you should define p outside the if

File: vochain/scrutinizer/process.go (hunk 2 above)
Comment (jordipainan) [reply vochain/scrutinizer/process.go:40]: agree, this is much more efficient

File: vochain/scrutinizer/process.go (hunk 3 above)
Comment (jordipainan) [reply vochain/scrutinizer/process.go:42]: same as before

**Comment by p4u** [comment]
Superseded by #110

**Pull request closed without merging**